Profiler/utils/Ipc.cpp

```diff
@@ -0,0 +1,145 @@
+#ifdef TIA
+#include "Ipc.h"
+
+#include <zmq.h>
+
+#define IPC_TIMEOUT_MS 250
+#define IPC_BUFFER_SIZE 255
+
+// ZMQ strings are not zero-terminated, let's terminate these
+#define IPC_TERMINATE_STRING(buffer, len) buffer[len < IPC_BUFFER_SIZE ? len : IPC_BUFFER_SIZE - 1] = '\0'
+
+Ipc::Ipc(Config* config, std::function<void(std::string)> testStartCallback, std::function<void(std::string, std::string)> testEndCallback, std::function<void(std::string)> errorCallback)
```

Comment:
There's a lot of magic happening in this class, w.r.t the number of `frames` and `len` of some return value. This is very hard to follow without a deeper knowledge of ZMQ. Maybe some explanatory comments would ease maintenance?